From 70add06dec8413965f3e98b6ba9a3c82168ea72c Mon Sep 17 00:00:00 2001
From: Hiroyuki Ikezoe <poincare@ikezoe.net>
Date: Tue, 16 Mar 2010 10:56:17 +0000
Subject: Fix crash.
Bug: https://bugzilla.gnome.org/show_bug.cgi?id=613022
Origin: http://git.gnome.org/browse/gpointing-device-settings/commit/?id=70add06dec8413965f3e98b6ba9a3c82168ea72c

We need to get xinput property if there is no gconf value for tapping time.

Fix for #613022.
---
--- a/modules/gpds-touchpad-ui.c
+++ b/modules/gpds-touchpad-ui.c
@@ -965,8 +965,14 @@
         }
     }
 
-    if (!gpds_ui_get_gconf_int(ui, GPDS_TOUCHPAD_TAP_TIME_KEY, &value))
+    if (!gpds_ui_get_gconf_int(ui, GPDS_TOUCHPAD_TAP_TIME_KEY, &value) && !values) {
+        if (!gpds_xinput_ui_get_xinput_int_property(GPDS_XINPUT_UI(ui),
+                                                    GPDS_TOUCHPAD_TAP_TIME,
+                                                    &values, &n_values)) {
+            return;
+        }
         value = values[0];
+    }
 
     double_value = value;
     object = gpds_ui_get_ui_object_by_name(GPDS_UI(ui), "tapping_time_scale");
